### ISSUE TITLE
It failed in the .net6 version, how to solve it? Version 4.7.2 successful

### ISSUE THREAD

**Issue by zhin2929** [camilochaves/MQL5_CSharp#3]
It failed in the .net6 version, how to solve it? Version 4.7.2 successful

`2024.08.12 17:56:45.912	CSharp_Net6 (XAUAUDm,M15)	function 'GuiMT_Net6.MQL5CSharp::ReadLastEvent' not found in 'D:\soft\MetaTrader 5\MQL5\Libraries\GuiMT_Net6.dll'`

**Comment by azhali** [comment]
hey @zhin2929 
Did you found the solution for making this work in .net 6?

**Comment by zhin2929** [comment]
> hey [@zhin2929](https://github.com/zhin2929) Did you found the solution for making this work in .net 6?

No

**Comment by darthlinuxer** [comment]
Yes. AI gave me these solutions but still need to test if they work

1. Use a TCP/IP Socket Connection

• MQL5 can send data to a .NET 6 application via sockets.

• Your .NET 6 application listens for messages and processes them.

• Example:

• MQL5 uses SocketCreate(), SocketSend(), and SocketClose().

• .NET 6 application uses System.Net.Sockets.TcpListener and TcpClient.


2. Use Named Pipes

• Named Pipes are a fast and efficient way to communicate between processes.

• .NET 6 supports System.IO.Pipes for this purpose.

• MQL5 can call a DLL that interacts with Named Pipes.


3. Use a Web API in .NET 6

• You can create a REST API in ASP.NET Core.

• MQL5 can make HTTP requests using WebRequest() or WebSocketCreate().


4. Use a Message Queue (MQTT, RabbitMQ, ZeroMQ)

• For real-time, asynchronous messaging, you can use an MQTT broker (e.g.,
Mosquitto) or ZeroMQ.


5. Use a Shared File or Database

• If real-time speed is not critical, MQL5 can write to a file or database,
and .NET 6 can process it.


Which One to Choose?

• Fast & Local Communication → Use Named Pipes or Sockets.

• Scalability & Remote Communication → Use a Web API.

• Async Messaging → Use MQTT or ZeroMQ.





Em dom., 16 de mar. de 2025 às 04:46, zhin ***@***.***>
escreveu:

**Comment by zhin2929** [comment]
> Yes. AI gave me these solutions but still need to test if they work
> 
> 1. Use a TCP/IP Socket Connection
> 
> • MQL5 can send data to a .NET 6 application via sockets.
> 
> • Your .NET 6 application listens for messages and processes them.
> 
> • Example:
> 
> • MQL5 uses SocketCreate(), SocketSend(), and SocketClose().
> 
> • .NET 6 application uses System.Net.Sockets.TcpListener and TcpClient.
> 
> 
> 2. Use Named Pipes
> 
> • Named Pipes are a fast and efficient way to communicate between processes.
> 
> • .NET 6 supports System.IO.Pipes for this purpose.
> 
> • MQL5 can call a DLL that interacts with Named Pipes.
> 
> 
> 3. Use a Web API in .NET 6
> 
> • You can create a REST API in ASP.NET Core.
> 
> • MQL5 can make HTTP requests using WebRequest() or WebSocketCreate().
> 
> 
> 4. Use a Message Queue (MQTT, RabbitMQ, ZeroMQ)
> 
> • For real-time, asynchronous messaging, you can use an MQTT broker (e.g.,
> Mosquitto) or ZeroMQ.
> 
> 
> 5. Use a Shared File or Database
> 
> • If real-time speed is not critical, MQL5 can write to a file or database,
> and .NET 6 can process it.
> 
> 
> Which One to Choose?
> 
> • Fast & Local Communication → Use Named Pipes or Sockets.
> 
> • Scalability & Remote Communication → Use a Web API.
> 
> • Async Messaging → Use MQTT or ZeroMQ.
> 
> 
> 
> 
> 
> Em dom., 16 de mar. de 2025 às 04:46, zhin ***@***.***>
> escreveu:
> […](#)

I use socket
